data-model/src/main/java/org/alfresco/service/cmr/repository/DirectAccessUrl.java

```diff
@@ -70,18 +71,28 @@ public void setAttachment(boolean attachment)
         this.attachment = attachment;
     }
 
+    public String getFileName()
+    {
+        return fileName;
+    }
+
+    public void setFileName(String fileName)
+    {
+        this.fileName = fileName;
+    }
+
     @Override public boolean equals(Object obj)
     {
         if (this == obj) return true;
         if (obj == null || getClass() != obj.getClass()) return false;
 
         DirectAccessUrl that = (DirectAccessUrl) obj;
-        return attachment == that.attachment && Objects.equals(contentUrl,
+        return fileName.equals(that.fileName) && attachment == that.attachment && Objects.equals(contentUrl,
```

Comment:
There is nothing preventing the `fileName` from being `null` so `NullPointerException` is possible here.
```suggestion
        return Objects.equals(fileName, that.fileName) && attachment == that.attachment && Objects.equals(contentUrl,
```